TODO/how-ios-apps-on-the-mac-could-work.md

```diff
@@ -1,112 +1,198 @@
+
 There are UIKit reimplementations like [TwUI](https://github.com/twitter/twui) and [Chameleon](http://chameleonproject.org), the latter seeking an identical API. In theory, you could share 100% of your UI code between platforms. In practice, these frameworks fight an uphill battle, because they third-party.
 
+当然也存在一些 UIKit 的重新实现的库，比如 [TwUI](https://github.com/twitter/twui) 和 [Chameleon](http://chameleonproject.org)，后者意在寻求相同的 API。理论上，你可以在不同的平台上共享 100% 的 UI 代码。但实际上，这些框架是往往是费力不讨好的，他们都是第三方的。
+
 Even among Apple’s Mac developers, there’s demand for a UIKit architecture. Last year’s Photos.app [included UXKit](https://sixcolors.com/post/2015/02/new-apple-photos-app-contains-uxkit-framework/), and Game Center uses a [UICollectionView replica](https://twitter.com/steipete/status/740065011712806912).
 
-### What I Expect
+即便是在 Mac 的开发者中，也有对 UIKit 架构的需求。去年，苹果官方的照片应用就包含了[UXKit](https://sixcolors.com/post/2015/02/new-apple-photos-app-contains-uxkit-framework/)，而游戏中心则采用了[UICollectionView](https://twitter.com/steipete/status/740065011712806912)做替代。
+
+### What I Expect 我的期望
 
 Don’t expect today’s iOS apps to launch on macOS without any changes. [Look at tvOS](https://developer.apple.com/library/tvos/documentation/General/Conceptual/AppleTV_PG/index.html#//apple_ref/doc/uid/TP40015241).
 
+不要期望现在的 iOS 不经改变就可以运行在 macOS 上面。看看 [tvOS](https://developer.apple.com/library/tvos/documentation/General/Conceptual/AppleTV_PG/index.html#//apple_ref/doc/uid/TP40015241)
+就知道了。
+
 > tvOS is derived from iOS but is a distinct OS, including some frameworks that are supported only on tvOS.
 
+> tvOS 就是 iOS 的一个衍生版本，但是却包含了很多只能在 tvOS 上使用的框架。
+
 It runs UIKit, with just enough stripped out to _force_ you to rewrite your interactions for a TV experience.
 
-#### One Bundle to Rule Them All?
+那上面也运行 UIKit，刚刚好能够让你重写一遍适合 TV 上的交互。
+
+#### One Bundle to Rule Them All? 只用写一个包 (Bundle) 就可以了？
 
 TV apps have such a different interaction model than a touch screen, you’ll probably end up with a radically different design.
 
+TV 上应用的交互方式和触摸屏上的方式差太多了，极有可能到最后，你会得到一个完全不同的设计。
+
 > When porting an existing project, you can include an additional target in your Xcode project to simplify sharing of resources, but you need to create new storyboards for tvOS. Likely, you will need to look at how users navigate through your app and adapt your app’s user interface to Apple TV.
 
+> 当移植现有的项目的时候，你可以附加一个目标在现有的 xcode 里面，简单的共享资源，但是你得为 tvOS 创建新的 storyboards。类似的，你需要研究用户如何使用你的应用，并且调整你的应用界面来适应 Apple TV。
+
 Even if you could bundle a tvOS app with your iOS app, the disadvantages (e.g. coupled releases) outweigh the advantages of shared code. I’ve known iOS developers who regretted shipped a universal iPhone/iPad app, because of the tight coupling. Sometimes you’re just better off just putting that shared code into a framework.
 
+即便你可以把 tvOS 应用和 iOS 应用放在一个 Bundle 里面，缺点（eg. 绑定发布？二次发布？）也大过优点。我知道不少的 iOS 的开发者都后悔发布了 iPhone/Ipad 上通用的应用，因为二者的联结太紧密。很多时候，倒不如把那些共享的代码放到一个框架（framework）里面。
```

Comment:
谢谢！我知道coupling 的意思，只是，我不知道coupled release 是什么意思，应该怎么翻译~~
在2016年06月30日 02:08，Wenlin Ou 写道:

In TODO/how-ios-apps-on-the-mac-could-work.md:

>  Even if you could bundle a tvOS app with your iOS app, the disadvantages (e.g. coupled releases) outweigh the advantages of shared code. I’ve known iOS developers who regretted shipped a universal iPhone/iPad app, because of the tight coupling. Sometimes you’re just better off just putting that shared code into a framework.
>  
> +即便你可以把 tvOS 应用和 iOS 应用放在一个 Bundle 里面，缺点（eg. 绑定发布？二次发布？）也大过优点。我知道不少的 iOS 的开发者都后悔发布了 iPhone/Ipad 上通用的应用，因为二者的联结太紧密。很多时候，倒不如把那些共享的代码放到一个框架（framework）里面。


维基百科：In software engineering, coupling is the manner and degree of interdependence between software modules; a measure of how closely connected two routines or modules are；the strength of the relationships between modules.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.